netbox/netbox_inventory.py

```diff
@@ -132,53 +182,84 @@ def fetch_inventory(cli_opts):
     if cli_opts.region:
         params["region"] = cli_opts.region
 
-    res = netbox.get("/api/dcim/devices/", params=params)
+    res = NETBOX_SESSION.get("/api/dcim/devices/", params=params)
```

Comment:
**Suggestion:** Implement pagination handling in the `fetch_inventory` function to manage large datasets that might exceed a single page of API results. [enhancement, importance: 9]
```suggestion
    devices = []
    while True:
        res = NETBOX_SESSION.get("/api/dcim/devices/", params=params)
        res.raise_for_status()
        data = res.json()
        devices.extend(data['results'])
        if 'next' not in data or data['next'] is None:
            break
        params['offset'] = data['next_offset']  # Assuming API response includes next_offset
    return devices
```